### PR TITLE
6847157: java.lang.NullPointerException: HDC for component at sun.java2d.loops.Blit.Blit

### PR DESCRIPTION
**Pull request by mrserb**
This is the fix for one suspicious exception which rarely but time to time occurred on the test systems. As described in a few bug reports the use-cases could be quite different, but I was able to reproduce it in one of them.

Steps to reproduce:
 1. We draw to the frame for the first time or we draw to the frame on the thread where we never draw before
 2. As part of step 1. we start initialization of the surface data bounded to this frame for that thread, in the native method
 3. jtreg decided to kill the test because the main method ends
 4. As part of step 2. we request HDC for the frame via sendMessage which do nothing since AWT is stopping/dies -> HDC=NULL
 5. As part of step 1. we try to draw to the surface -> HDC is NULL -> NPE is thrown

The solution is to mark the surface at step 4. as invalid:
 - If the AWT will die this surface will never be used and no exception occurs
 - If the application will continue to work and this surface will be accessed then we will throw InvalidPipeException in attempts to recreate the surface and all related data.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x32 | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |  ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-6847157](https://bugs.openjdk.java.net/browse/JDK-6847157): java.lang.NullPointerException: HDC for component at sun.java2d.loops.Blit.Blit

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/870/head:pull/870`
`$ git checkout pull/870`

### CODE REVIEW

File: test/jdk/java/awt/Paint/RepaintOnAWTShutdown.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @run main/othervm                           RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.2  RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.25 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.5  RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.75 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=2    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=2.25 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=5    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=10    RepaintOnAWTShutdown
```
Comment (mrserb) [inline test/jdk/java/awt/Paint/RepaintOnAWTShutdown.java:51]: The bug depends on the CPU/GPU performance so I tried to reproduce it by tweaking the number of pixels to draw.

File: test/jdk/java/awt/Paint/RepaintOnAWTShutdown.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * @run main/othervm                           RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.2  RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.25 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.5  RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=1.75 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=2    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=2.25 RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=5    RepaintOnAWTShutdown
+ * @run main/othervm -Dsun.java2d.uiScale=10   RepaintOnAWTShutdown
```
Comment (mrserb) [inline test/jdk/java/awt/Paint/RepaintOnAWTShutdown.java:51]: The bug depends on the CPU/GPU performance so I tried to reproduce it by tweaking the number of pixels to draw.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**